tests/test.py

```diff
@@ -512,6 +514,457 @@ def testZstashWithHPSS(self):
+        self.str_not_in(cmd, hpss_path, output+err, 'ERROR')
+        cmd = 'zstash check -v --hpss={}'.format(hpss_path)
+        output, err = run_cmd(cmd)
+        self.str_in(cmd, hpss_path, output+err, 'Checking file0.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Checking file0_hard.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Checking file0_soft.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Checking file_empty.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Checking dir/file1.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Checking empty_dir')
+        self.str_in(cmd, hpss_path, output+err, 'Checking dir2/file2.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Checking file3.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Checking file4.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Checking file5.txt')
+        self.str_not_in(cmd, hpss_path, output+err, 'ERROR')
+
+        print('8. Testing the extract functionality')
+        os.chdir('../')
+        os.rename('zstash_test', 'zstash_test_backup')
+        os.mkdir('zstash_test')
+        os.chdir('zstash_test')
+        shutil.copytree('../zstash_test_backup/zstash', 'zstash')
+        cmd = 'zstash extract --hpss={}'.format(hpss_path)
+        output, err = run_cmd(cmd)
+        os.chdir('../')
+        self.str_in(cmd, hpss_path, output+err, 'Extracting file0.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Extracting file0_hard.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Extracting file0_soft.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Extracting file_empty.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Extracting dir/file1.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Extracting empty_dir')
+        self.str_in(cmd, hpss_path, output+err, 'Extracting dir2/file2.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Extracting file3.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Extracting file4.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Extracting file5.txt')
+        self.str_not_in(cmd, hpss_path, output+err, 'ERROR')
+
+        print('9. Testing the extract functionality again, nothing should happen')
+        os.chdir('zstash_test')
+        cmd = 'zstash extract -v --hpss={}'.format(hpss_path)
+        output, err = run_cmd(cmd)
+        os.chdir('../')
+        self.str_in(cmd, hpss_path, output+err, 'Not extracting file0.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Not extracting file0_hard.txt')
+        # It's okay to extract the symlinks.
+        self.str_not_in(cmd, hpss_path, output+err, 'Not extracting file0_soft.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Not extracting file_empty.txt')
+        self.str_in(cmd, hpss_path, output+err, 'Not extracting dir/file1.txt')
+        # It's okay to extract empty dirs.
+        #self.str_not_in(cmd, hpss_path, output+err, 'Not extracting empty_dir')
```

Comment:
Interestingly, this fails on Compy, but not Cori. If we're ok with having "not extracting empty_dir" in `output+err`, then I can remove this commented assertion.